sync-gateway-config.json

```diff
@@ -97,10 +96,12 @@ function(doc, oldDoc){
     access(doc.owner, "task-list." + doc._id + ".users");
   } else if (getType() == "task") {
     /* Write Access */
+    var owner = doc._deleted ? oldDoc.taskList.owner : doc.taskList.owner;
```

Comment:
The presence of `taskList` is checked later on at https://github.com/couchbaselabs/mobile-training-todo/blob/master/sync-gateway-config.json#L109

I'm wondering if it'd make more sense to do all schema validation before write permissions?

The reason for having write permissions before validation was to minimize the CPU cost for rejected documents. That if the user's got no write permission there's no point in checking the schema. But equally, if the schema isn't valid then there's no point in checking for write permission.

It looks like doing validation before write permissions would solve the issue here. The flow would change from:

```
Write permission -> validation -> routing -> read permission
```

to 

```
Validation -> write permission -> routing -> read permission
```

Let me know if you agree with this general change and I'll update the sync function with the new ordering for all doc types.